Update taint feature design doc and user manual (#1632)
FoundationDB/fdb-kubernetes-operator: docs/manual/replacements_and_deletions.md

The number of inflight replacements can be configured by setting `maxConcurrentReplacements`, per default the operator will replace all misconfigured process groups.
Depending on the cluster size this can require a quota that is has double the capacity of the actual required resources.

> ## Automatic Replacements for Pods in Bad State

Reviewer: ```suggestion
## Automatic Replacements for ProcessGroups in Bad State
```

Please use the correct term in the documentation, the operator acts on ProcessGroups and the Pods are only one part of it. What is the benefit of adding `for ProcessGroups in Bad State`? What else should the operator automatically be replacing?

Author: It's to distinguish from Replacements for Pods on Tainted Nodes.

> ## Automatic Replacements for ProcessGroups in Bad State

Author: I changed it as well.

* `MissingPod`: This indicates a process group that doesn't have a Pod assigned.
* `MissingPVC`: This indicates that a process group that doesn't have a PVC assigned.
* `MissingService`: This indicates that a process group that doesn't have a Service assigned.
* `PodPending`: This indicates that a process group where the Pod is in a pending state.

> * `NodeTaintReplacing`: This indicates that a process group where the pod is running on a tainted Node

Reviewer: ```suggestion
* `NodeTaintReplacing`: This indicates that a process group where the pod is running on a tainted Node.
```

This is only half true, if the node is not tainted anymore the ProcessGroup will be replaced anyway (at least in the current implementation) if the `NodeTaintReplacing` condition is set. We should add this details here. The implication there is that if a ProcessGroup has the `NodeTaintReplacing` condition the replacement cannot be stopped, even if the node was fixed.

Process groups that are set into the crash loop state with the `Buggify` setting won't be replaced by the operator.
If the `cluster.Spec.Buggify.EmptyMonitorConf` setting is active the operator won't replace any process groups.

> ## Automatic Replacements for Pods on Tainted Nodes
> The operator has an option to automatically replace Pods that are running on tainted Nodes. This feature is disabled by default but can be enabled by setting `automationOptions.replacements.taintReplacementOptions` and `automationOptions.replacements.taintReplacementTimeSeconds`.

Reviewer: ```suggestion
## Automatic Replacements for Pods on Tainted Nodes

The operator has an option to automatically replace ProcessGroups where the associated Pod is running on a tainted Node. This feature is disabled by default, but can be enabled by setting `automationOptions.replacements.taintReplacementOptions`.
```

`automationOptions.replacements.taintReplacementTimeSeconds` is actually an optional setting. We should remove this here.

> **Example 1.** We can use the following setup to let operator detect Pods running on Nodes with taint key `example.com/maintenance`, set the Pods' condition to `NodeTaintReplacing` if their Nodes have been tainted for 3600 seconds, and replace the Pods after 1800 seconds.

Reviewer: ```suggestion
### Example 1

The following setup lets the operator detect Pods running on Nodes with taint key `example.com/maintenance`, set the ProcessGroup' condition to `NodeTaintReplacing`, if their Nodes have been tainted for 3600 seconds, and replace the Pods after 1800 seconds.
```

Can we use proper markdown syntax and maybe a better/more precise heading than `example 1`? I personally find it a little bit hard to read this sentence.

```
automationOptions.replacements.taintReplacementOptions = {
    {
        Key: example.com/maintenance
        DurationInSeconds: 3600
    },
}
automationOptions.replacements.taintReplacementTimeSeconds = 1800

> ```

Author: changed it to yaml format

> If there are multiple Pods on tainted Nodes, the operator will simultaneously replace at most `automationOptions.replacements.maxConcurrentReplacements` Pods.

Author: the sentence above is in a parallel section "Automatic Replacements for ProcessGroups in Bad State". 
Having this sentence here is to explicitly state the option is also in effect for "Automatic Replacements for ProcessGroups on Tainted Nodes".
It helps to avoid ambiguity to readers.

> **Example 2.** We can enable taint feature on all taint keys except one taint key with the following configuration:

Author: Changed it to 
```

We use three examples below to illustrate how to set up the feature.

### Example Setup 1
...
### Example Setup 2
...
### Example Setup 3
...
```

```
automationOptions.replacements.taintReplacementOptions = {
    {
		Key:               *,
		DurationInSeconds: 3600,
	},
	{
		Key:               example.com/taint-key-to-ignore,
		DurationInSeconds: 9223372036854775807,
	},
}
```

> The operator will detect and mark all Pods on tainted Nodes with `NodeTaintDetected` condition. But the operator will not mark Pods on Nodes with taint key `example.com/taint-key-to-ignore` with `NodeTaintReplacing` condition because the key's `DurationInSeconds` is set to max of int64.

Author: `
The operator will detect and mark all Pods on tainted Nodes with `NodeTaintDetected` condition. But the operator will ignore the taint key `example.com/taint-key-to-ignore` when it adds `NodeTaintReplacing` condition to Pods, because the key's `DurationInSeconds` is set to max of int64. For example, if a Node has only the taint key `example.com/taint-key-to-ignore`, its Pods will only be marked with  `NodeTaintDetected` condition. When the Node has another taint key, say `example.com/any-other-key`, its Pods will be added `NodeTaintReplacing` condition when the other taint key has been on the Node for 3600 seconds.
`

> The operator will detect and mark all Pods on tainted Nodes with `NodeTaintDetected` condition. But the operator will not mark Pods on Nodes with taint key `example.com/taint-key-to-ignore` with `NodeTaintReplacing` condition because the key's `DurationInSeconds` is set to max of int64.

Author: changed to this ^^^
